keras_nlp/utils/text_generation.py

```diff
@@ -128,29 +132,54 @@ def token_probability_fn(inputs):
     ```
 
     """
-    if not tf.executing_eagerly():
-        raise RuntimeError(
-            "`keras_nlp.utils.greedy_search` currently requires an eager "
-            "execution context. Please call `greedy_search` outside "
-            "tf.function or run `tf.config.run_functions_eagerly(True)` to run "
-            "tf.function in eager mode."
-        )
-
     prompt = validate_prompt(prompt)
 
     input_is_1d = prompt.shape.rank == 1
     if input_is_1d:
         prompt = prompt[tf.newaxis, :]
     validate_token_probability_fn(token_probability_fn, prompt)
 
-    i = prompt.shape[1]
-    while i < max_length:
-        # If the prompt has reached our desired length, exit while loop.
-        pred = token_probability_fn(prompt)
+    length = prompt.shape.as_list()[1]
```

Comment:
can we just do something like
```
batch_size, length = tf.shape(x)
```
And use that below? Then length and batch size are both tensors from the start.

Reply:
Hmmm, I'll split this into two lines:

```
batch_size = tf.shape(prompt)[0]
length = tf.shape(prompt)[1]
```

```
batch_size, length = tf.shape(x)
```
does not work in graph mode.

Reply:
Stack trace: https://p.ip.fi/6YAg

Reply:
ah destructuring is too fancy for autograph, I forgot. let's do

shape = tf.shape(prompt)
batch_size = shape[0]
length = shape[1]